flow/flow.go

```diff
@@ -170,7 +179,11 @@ func addGenerator(out low.Rings, generateFunction GenerateFunction, context User
 	par.generateFunction = generateFunction
 	ctx := make([]UserContext, 1, 1)
 	ctx[0] = context
-	schedState.addFF("generator", nil, nil, pGenerate, par, &ctx, generate, 0)
+	name := schedState.addFF("generator", nil, nil, pGenerate, par, &ctx, generate, 0)
+	if countersEnabled {
```

Comment:
I propose to put register inside addFF. Scheduler is in the same packet - shouldn't be a problem. Otherwise a lot of identical stuff.